src/hb-bit-page.hh

```diff
@@ -86,6 +86,58 @@ struct hb_bit_page_t
   void set_range (hb_codepoint_t a, hb_codepoint_t b, bool v)
   { if (v) add_range (a, b); else del_range (a, b); }
 
+
+  // Writes out page values to the array p. Returns the number of values
+  // written. At most size codepoints will be written.
+  unsigned int export_array (uint32_t base, hb_codepoint_t *p, unsigned int size) const
+  {
+    unsigned int count = 0;
+    for (unsigned i = 0; i < len () && count < size; i++)
+    {
+      elt_t bits = v[i];
+      uint32_t v_offset = i << ELT_BITS_LOG_2;
+      for (unsigned int j = 0; j < ELT_BITS && count < size; j++)
+	if ((elt_t (1) << j) & bits)
+	{
+	  *p++ = base | v_offset | j;
```

Comment:
Any reason you don't add `base` to `v_offset` outside the loop?

Reply:
Good idea. Done.